rabbit-escape-render/src/rabbitescape/render/WaterRegionRenderer.java

```diff
@@ -43,21 +29,23 @@
     private final WaterAnimation waterAnimation;
 
     public WaterRegionRenderer(
-        WaterRegion region, 
+        int x,
+        int y,
```

Comment:
This is the bulk of the change in the PR. Replace a copy of the WaterRegion (which sometimes gets outdated and so the graphics go weird) with coordinates so the WaterRegion can be looked up.